File: mobile/src/main/kotlin/de/scout/fireplace/settings/SimpleTextWatcher.java
package de.scout.fireplace.settings;

import android.text.Editable;
import android.text.TextWatcher;

public class SimpleTextWatcher implements TextWatcher {

  private final TextChangedListener listener;

  public SimpleTextWatcher(TextChangedListener listener) {
    this.listener = listener;
  }

  @Override
  public void beforeTextChanged(CharSequence sequence, int start, int count, int after) {
    /* no-op */
  }

  @Override
  public void onTextChanged(CharSequence sequence, int start, int count, int after) {
    listener.onTextChanged(sequence.toString());
  }

  @Override
  public void afterTextChanged(Editable editable) {
    /* no-op */
  }

  public interface TextChangedListener {

    void onTextChanged(String value);
  }
}

File: feature/src/main/kotlin/de/scout/fireplace/feature/settings/SimpleTextWatcher.kt
package de.scout.fireplace.feature.settings

import android.text.Editable
import android.text.TextWatcher

internal class SimpleTextWatcher(private val function: (String) -> Unit) : TextWatcher {

  override fun beforeTextChanged(sequence: CharSequence, start: Int, count: Int, after: Int) = Unit

  override fun onTextChanged(sequence: CharSequence, start: Int, count: Int, after: Int) = function(sequence.toString())

  override fun afterTextChanged(editable: Editable) = Unit
}
